pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -829,9 +864,112 @@ func (n *NetworkPolicyController) deletePod(old interface{}) {
 
 	klog.V(2).Infof("Processing Pod %s/%s DELETE event, labels: %v", pod.Namespace, pod.Name, pod.Labels)
 	// Find all AppliedToGroup keys which match the Pod's labels.
-	appliedToGroupKeys := n.filterAppliedToGroupsForPod(pod)
+	appliedToGroupKeys := n.filterAppliedToGroupsForPodOrExternalEntity(pod)
+	// Find all AddressGroup keys which match the Pod's labels.
+	addressGroupKeys := n.filterAddressGroupsForPodOrExternalEntity(pod)
+	// Enqueue groups to their respective queues for group processing.
+	for group := range appliedToGroupKeys {
+		n.enqueueAppliedToGroup(group)
+	}
+	for group := range addressGroupKeys {
+		n.enqueueAddressGroup(group)
+	}
+}
+
+// addExternalEntity retrieves all AddressGroups and AppliedToGroups which match the ExternalEnitty's
+// labels and enqueues the groups key for further processing.
+func (n *NetworkPolicyController) addExternalEntity(obj interface{}) {
+	defer n.heartbeat("addExternalEntity")
+	ee := obj.(*v1alpha1.ExternalEntity)
+	klog.V(2).Infof("Processing External Entity %s/%s ADD event, labels: %v", ee.Namespace, ee.Name, ee.Labels)
+	// Find all AppliedToGroup keys which match the ExternalEntity's labels.
+	appliedToGroupKeySet := n.filterAppliedToGroupsForPodOrExternalEntity(ee)
+	// Find all AddressGroup keys which match the ExternalEntity's labels.
+	addressGroupKeySet := n.filterAddressGroupsForPodOrExternalEntity(ee)
+	// Enqueue groups to their respective queues for group processing.
+	for group := range appliedToGroupKeySet {
+		n.enqueueAppliedToGroup(group)
+	}
+	for group := range addressGroupKeySet {
+		n.enqueueAddressGroup(group)
+	}
+}
+
+// updateExternalEntity retrieves all AddressGroups and AppliedToGroups which match the
+// updated and old ExternalEntity's labels and enqueues the group keys for further
+// processing.
+func (n *NetworkPolicyController) updateExternalEntity(oldObj, curObj interface{}) {
```

Comment:
would also prefer to move out ee related methods to its own .go file so that this file does not grow out of control

Reply:
Right now I don't feel like EE add/update/delete is much different than those of pod/namespace. If we want to move ee related methods to a separate file maybe it would make sense to move pod processing methods as well? We can revisit this if there are actually more functions related to EE being added